ytt/001-schemas/README.md

```diff
@@ -239,111 +900,393 @@ Beyond specifying a type for a value, one can specify more dynamic constraints o
   - base64.decodable()
   - json.decodable()
   - yaml.decodable()
+  - regexp.is_valid()
 
   String values, by default, have `@schema/validate min_len=1` validation added.
 
-### Describing annotations
+## Part 8: Interactions with Existing Features
 
-#### `@schema/title`
+### Changes to Data Values
 
-  This annotation provides a way to add a short title to the section of a schema associated with a key, similar to the [JSON schema title field](https://json-schema.org/draft/2019-09/json-schema-validation.html#rfc.section.9.1).
+Prior to Schemas, Data Values protected against mistakes like typo'ed key names through the
+strict merge of additional data values documents: to successfully merge a new key, it must be marked
+as `@overlay/match missing_ok=True`.
 
-  ```yaml
-  #@schema/title "User Password"
-  user_password: ""
-  ```
+The responsibility to ensure key names are valid are shifted from Data Values to Schema.
 
-  If the annotation is not present, the title will be inferred from the key name by replacing special characters with spaces and capitalizing the first letter similar to [rails humanize functionality](https://apidock.com/rails/String/humanize).
+When schema is present, Data Values documents imply an automatic merge of new keys. This is 
+equivalent to all `@data/values` documents being annotated with `@overlay/match-child-defaults missing_ok=True`.
 
-#### `@schema/doc`
+### Overlays of Schema
 
-  This annotation is a way to add a longer description to the section of a schema associated with a key, similar to the JSON Schema description field.
+TBD
 
-  ```yaml
-  #@schema/doc "The user password used to log in to the system"
-  user_password: ""
-  ```
+- It must be possible to extend Schema (i.e. add new items)
+- Is it necessary to be able to modify Schema?
 
-#### `@schema/example`
 
-  This annotation will take one argument that is an example of a value that satisfies the schema.
+## Open Questions
 
-  ```yaml
-  #@schema/example "my_domain.example.com"
-  system_domain: ""
-  ```
+- It is not uncommon for a validation to want to refer to a sibling node.
+    - right now, the Configuration Author places these on the parent node.
+    - that puts separation between the node being constrained and its constraint.
+    - is there a feasible way for a validation rule to get access to its siblings?
+- Documentation could be enhanced if validation rules were included. How could validation authors
+  provide documentation for their validation?
+  - provide a magic string for the input and the function returns documentation?
+  - function always returns documentation as part of return value?
+- How should schema work with ytt libraries?
+  - How do we support data values that are targetting a dependency library?
+- Are there needs around versioning schema?
```

Comment:
As described above with the exploration of the `version=` keyword argument of `@schema/definition`: no.